src/hub/views.py

```diff
@@ -504,6 +508,89 @@ def _get_latest_actions_context(self):
         }
         return context
 
+    @action(
+        detail=False,
+        methods=[GET],
+        permission_classes=[AllowAny]
+    )
+    def by_contributions(self, request, pk=None):
+        hub_id = request.GET.get('hub_id', None)
+        hub_qs = Hub.objects.all().distinct() if hub_id is None \
+            else Hub.objects.filter(id=hub_id)
```

Comment:
[Line continuations are ugly.](https://www.python.org/dev/peps/pep-0008/)

```suggestion
        hub_id = request.query_params.get('hub_id', None)
        hub_qs = self.queryset
        if hub_id:
            hub_qs = hub_qs.filter(id=hub_Id)
```